sensor/debugger/k8s/fake.go

```diff
@@ -0,0 +1,271 @@
+		clusterRoleKind:           &rbacv1.ClusterRole{},
+		roleBindingKind:           &rbacv1.RoleBinding{},
+		clusterRoleBindingKind:    &rbacv1.ClusterRoleBinding{},
+		networkPolicyKind:         &networkingv1.NetworkPolicy{},
+		nodeKind:                  &corev1.Node{},
+		serviceKind:               &corev1.Service{},
+		jobKind:                   &batchv1.Job{},
+		replicaSetKind:            &appsv1.ReplicaSet{},
+		replicationControllerKind: &corev1.ReplicationController{},
+		daemonSetKind:             &appsv1.DaemonSet{},
+		deploymentKind:            &appsv1.Deployment{},
+		statefulSetKind:           &appsv1.StatefulSet{},
+		cronJobKind:               &batchv1.CronJob{},
+		podKind:                   &corev1.Pod{},
+	}
+}
+
+// waitForMinimumResources waits for a minimum number of resources to be created or once all the events have been processed
+func waitForMinimumResources(ch chan string, done concurrency.Signal) error {
+	count := 0
+	doneC := done.WaitC()
+	for {
+		select {
+		case obj := <-ch:
+			if _, ok := minimumResources[obj]; ok {
+				minimumResources[obj]--
+				if minimumResources[obj] == 0 {
+					count++
+				}
+				if len(minimumResources) == count {
+					return nil
+				}
+			}
+		case <-doneC:
+			return errors.New("the events file did not contain the minimum resources required to start sensor")
+		}
+	}
+}
+
+// CreateEvents creates the k8s events from a given jsonl file
+func (f *FakeEventsManager) CreateEvents() error {
+	ch := make(chan string)
+	done := concurrency.NewSignal()
+	objs, err := f.Reader.ReadFile()
+	if err != nil {
+		return err
+	}
+	f.Init()
+	go func() {
+		for _, obj := range objs {
+			if len(obj) == 0 {
+				continue
+			}
+			msg := resources.InformerK8sMsg{}
+			if err := json.Unmarshal(obj, &msg); err != nil {
+				log.Fatalln(err)
+			}
+			log.Printf("%s Event: %s", msg.Action, msg.ObjectType)
+			if err := f.createEvent(msg, ch); err != nil {
+				log.Fatalf("cannot create event for %s: %s", msg.ObjectType, err)
+			}
+		}
+		done.Signal()
+	}()
+	return waitForMinimumResources(ch, done)
+}
+
+// runOp runs the create/update/delete operation
+func runOp(action string, resourceClient, resourceObject reflect.Value) []reflect.Value {
+	method, ok := actionToMethod[action]
+	if !ok {
+		log.Fatalf("method not found")
+	}
+	options, ok := actionToOptions[action]
+	if !ok {
+		log.Fatalf("options not found")
+	}
+	return resourceClient.MethodByName(method).Call([]reflect.Value{
+		reflect.ValueOf(context.Background()),
+		resourceObject,
+		reflect.ValueOf(options),
+	})
+}
+
+// getNamespace returns the namespace from a resource
+func getNamespace(resource reflect.Value) string {
+	values := resource.MethodByName("GetNamespace").Call([]reflect.Value{})
+	if len(values) != 1 {
+		return ""
+	}
+	return values[0].String()
+}
+
+// handleRunOp handles the execution of runOp
+func (f *FakeEventsManager) handleRunOp(action, kind string, client, object reflect.Value, ch chan string) error {
```

Comment:
Following on my previous comment, this function is a writer so it should create the channel inside and return it to the caller.